src/xcode/ENA/ENA/Source/Scenes/Events/Store/Model/TraceLocation.swift

```diff
@@ -47,7 +47,8 @@ struct TraceLocation: Equatable {
 			let eventDuration = Calendar.current.dateComponents(
 				[.minute],
 				from: startDate,
-				to: endDate
+				// The event duration should be atleast 15 minutes
+				to: endDate.addingTimeInterval(15 * 60)
 			).minute
 
 			duration = eventDuration ?? fallback
```

Comment:
At the moment you'll add 15 minutes everytime for `suggestedCheckoutLengthInMinutes: Int`. I guess that's not what you would like to have. 

Rather a reaction, when startDate and endDate are the same (length = 0). Then the fallback would be `15` minutes.
That's also what I've been implemented with [14148](https://github.com/corona-warn-app/cwa-app-ios/pull/4956/files) (you can compare with the commit history). But it seems, the the change was overwritten.

Could you please check the App behaviour with that adoption:

```swift

let eventDuration = Calendar.current.dateComponents(
	[.minute],
	from: startDate,
	to: endDate
).minute

duration = eventDuration.map { $0 == 0 ? fallback : $0 } ?? fallback // <-- HERE
```

Reply:
fallback adjusted